### PR TITLE
Superfluid features

### PR DESCRIPTION
**Pull request by Lewington-pitsos**
Database now tracks "Gateway" objects (see `Database.sol`)

Gateways store the addresses of all users checked in to them. Gateways have a single payee address.

Using this information, the frontend can collect all flows currently targeting a payee address, remove all those that are not currently active in a gateway, and achieve a total current flows for that gateway.

Gateways always set their payee as the address which sent the `addGate` transaction. If a merchant wants to create a gateway whose payee is another address, they will have to log out and log back in again using that new address.

Gateway Deletions will be added in the next PR.

### CODE REVIEW

**Comment by oschleic** [comment]
> * **How do we decide the flow rate?**

I think it would be best for a user to set the flow rate for each client they create. Possibly change the current client database from

```
mapping (uint256 => address) private clientIds;
```

to something like

```
struct Client{
bytes name;
uint flowrate;
address _address;
};

mapping(uint256 => Client) private clientIds;
```

**Comment by oschleic** [comment]
Also think this can probably be removed. uint256 allows for plenty of clientID creations and users can always make another wallet anyways
https://github.com/Laminar-Finance/Laminar-Core/blob/3637c27129d9857cc89a24e876e86d080db846bf/contracts/PaymentReceiver.sol#L16

**Comment by Lewington-pitsos** [comment]
If we have an external database anyway I think we need to store as little as possible on the blockchian because (1) costs (2) managing data on the blockchain is a huge pain in the ass and we _will_ time if we can manage all the required queries/updates through another framework.

Ideally we won't need one though.

**Comment by oschleic** [comment]
> Ideally we won't need one though.

Yeah trying to avoid this if possible. We'll for sure need to be able to edit the flow rate through the contract and a name will also be helpful if the user wants to edit the flow rate or delete a specific client.

**Comment by oschleic** [comment]
Also thoughts on renaming 'client' to something else? I find it confusing and think gate/gateID would be more fitting.

**Comment by 2manslkh** [comment]
I would say that `client` is more general as it our service doesn't just have to be fitted onto `gates`

File: hardhat.config.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 require("dotenv").config();
 
+require("@nomiclabs/hardhat-truffle5");
```
Comment (2manslkh) [inline hardhat.config.js:3]: Why need hardhat truffle

Comment (2manslkh) [reply hardhat.config.js:3]: Other than that LGTM

Comment (Lewington-pitsos) [reply hardhat.config.js:3]: it is required for importing the `web3` package. Without this import I get errors like`Error: A web3 instance is not provided when not using truffle.`

**Pull request merged**